scripts/generate_embeddings.py

```diff
@@ -116,18 +125,30 @@ def got_whitespace(text: str) -> bool:
     vector_store = FaissVectorStore(faiss_index=faiss_index)
     storage_context = StorageContext.from_defaults(vector_store=vector_store)
 
+    # Load documents
     documents = SimpleDirectoryReader(
         args.folder, recursive=True, file_metadata=file_metadata_func
     ).load_data()
 
-    good_nodes = []
+    # Split based on header/section
+    md_parser = MarkdownNodeParser()
+    documents = md_parser.get_nodes_from_documents(documents)
+
+    # Create chunks/nodes
     nodes = Settings.text_splitter.get_nodes_from_documents(documents)
+
```

Comment:
I'm not sure I understand the role of  MarkdownNodeParser here. How is this different from simply changing the chunk size to 380 and overlap to 0?

Reply:
I'm wondering if sections by themselves can be chunks, instead of slicing into smaller chunks and joining together top N of them. Now that we have a way of iterating over sections, we can find out the distribution of their sizes in tokens.